client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -504,4 +552,38 @@ public void disconnect() {
         }
         return check;
     }
+
+    private <V> V winrmCallRetryConnFailure(CallableFunction<V> winrmCall) throws SOAPFaultException {
+        int retries = retriesForConnectionFailures != null ? retriesForConnectionFailures : 10;
+        List<Throwable> exceptions = new ArrayList<>();
+
+        for (int i = 0; i < retries; i++) {
+            try {
+                return winrmCall.call();
+            } catch (SOAPFaultException soapFault) {
+                throw soapFault;
+            } catch (javax.xml.ws.WebServiceException wsException) {
+                if (wsException.getCause() instanceof ConnectException) {
+                    if (i == (retries + 1)) {
+                        LOG.info("Propagating Connection RM exception (attempt " + (i + 1) + " of " + retries + ") {}", wsException);
+                    } else if (i == 0) {
+                        LOG.warn("Ignoring exception and retrying (attempt " + (i + 1) + " of " + retries + ") {}", wsException);
+                    } else {
+                        LOG.debug("Ignoring exception and retrying (attempt " + (i + 1) + " of " + retries + ") {}", wsException);
+                    }
+                }
+                exceptions.add(wsException);
+            }
+        }
+        throw new RuntimeException("failed task "+winrmCall, exceptions.get(exceptions.size()-1));
```

Comment:
Would it be better to get the first exception as it might contain the most useful info for the cause of the failure?